File: php/diamond/Diamond.php
<?php

function diamond(string $letter) {
    $distance = ord($letter) - ord('A');
    $res = [];
    for ($i = 0; $i <= $distance; ++$i) {
        $res[$i] = str_repeat(' ', $distance - $i) . chr(ord('A') + $i) . str_repeat(' ', $i);
        $res[$i] .= substr(strrev($res[$i]), 1);
    }
    for ($i = $distance - 1; $i >= 0; --$i) {
        $res[] = $res[$i];
    }
    return $res;
}

File: nim/diamond/diamond.nim
import sequtils, strutils

proc diamond*(letter: char): string =
  let side = (ord(letter) - ord('A')) * 2 + 1
  var square = sequtils.repeat(spaces(side), side)
  for c in 'A'..letter:
    let
      charIdx = ord(c) - ord('A')
      x1 = int((side - 1) / 2) - charIdx
      x2 = int((side - 1) / 2) + charIdx
      y1 = charIdx
      y2 = side - 1 - charIdx
    square[y1][x1] = c
    square[y2][x1] = c
    square[y1][x2] = c
    square[y2][x2] = c
  square.join("\n") & "\n"
